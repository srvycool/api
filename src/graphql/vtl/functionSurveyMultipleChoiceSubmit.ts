export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "UpdateItem",
    "key" : {
      "node" : $util.dynamodb.toDynamoDBJson($ctx.stash.surveyID),
      "path" : $util.dynamodb.toDynamoDBJson("answer#$ctx.stash.answerID")
    },
    "update" : {
      "expression" : "ADD #votes :one",
      "expressionNames": {
        "#votes" : "votes"
      },
      "expressionValues": {
        ":one" : { "N": 1 }
      }
    }
  }
`;

export const response = `
  {}
`;
