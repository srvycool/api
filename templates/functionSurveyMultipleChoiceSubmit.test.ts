import Parser, { Context } from "appsync-template-tester";
import { request, response } from "./functionSurveyMultipleChoiceSubmit";

describe("templates:functionSurveyMultipleChoiceSubmit", () => {
  describe("request", () => {
    it("should match snapshot", () => {
      const parser = new Parser(request);
      const context = {
        stash: {
          surveyID: "aa-bb-cc",
          answerID: "dd-ee-ff",
        },
      } as Context;

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe("response", () => {
    it("should match snapshot", () => {
      const parser = new Parser(response);
      const context = {};

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});
