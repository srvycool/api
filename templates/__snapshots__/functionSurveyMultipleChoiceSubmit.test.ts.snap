// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates:functionSurveyMultipleChoiceSubmit request should match snapshot 1`] = `
Object {
  "key": Object {
    "node": Object {
      "S": "aa-bb-cc",
    },
    "path": Object {
      "S": "answer#dd-ee-ff",
    },
  },
  "operation": "UpdateItem",
  "update": Object {
    "expression": "ADD #votes :one",
    "expressionNames": Object {
      "#votes": "votes",
    },
    "expressionValues": Object {
      ":one": Object {
        "N": 1,
      },
    },
  },
  "version": "2017-02-28",
}
`;

exports[`templates:functionSurveyMultipleChoiceSubmit response should match snapshot 1`] = `Object {}`;
